packages/core/src/sync-historical/service.ts

```diff
@@ -198,12 +197,10 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
 
     await Promise.all(
       this.sources.map(async (source) => {
-        const { isHistoricalSyncRequired, startBlock, endBlock } =
-          getHistoricalBlockRange({
-            startBlock: source.startBlock,
-            endBlock: source.endBlock,
-            finalizedBlockNumber,
-          });
+        const startBlock = source.startBlock;
+        const endBlock = source.endBlock ?? finalizedBlockNumber;
+        const isHistoricalSyncRequired =
```

Comment:
We might be able to go one step further and consolidate the `if (!isHistoricalSyncRequired) { ...` logic from each branch below